### PR TITLE
Test for quickly running unit tests in the PR-CI build (with logging output).

### PR DESCRIPTION
**Pull request by d3r3kk**
Test for mocha vsts reporter...

### CODE REVIEW

**Comment by d3r3kk** [comment]
@DonJayamanne please take a look at the logs in the CI-LinuxShenanigans build in the checks. The logs are being output to the VSTS console in a way that looks good to me, but please have a look to see if you agree or not. 
I also simply 'extended' the functionality of the mocha-junit-reporter instead of incorporating it into our codebase. I don't like how I've 'imported' it in the unittests class, if you have any suggestions on improving that (without necessarily adding it to our package?) that would be aces.

**Comment by d3r3kk** [comment]
If you like the console output the way I have it, I'll try and get it incorporated to all the tests as soon as I can get to it.
But I'll definitely get unittests running in the initial phase of our CI builds to give us an 'early exit' (and to provide a clean and quick feedback mechanism for further unit tests!).

**Comment by DonJayamanne** [comment]
The PR is in the wrong repo. Time to sleep...lol

**Comment by d3r3kk** [comment]
@DonJayamanne yep, intended that way (but *not* intended to block your sleep time!). I'll move it over to the main repo if you agree with the console output seen here.

**Comment by DonJayamanne** [comment]
Will check now

File: src/test/unittests.ts [hunk 1]
```diff
@@ -12,20 +12,44 @@ import * as glob from 'glob';
...
+
     const options: MochaSetupOptions = {
         ui: 'tdd',
         useColors: true,
         timeout,
         grep
     };
-    const mocha = new Mocha(options);
+
+    let temp_mocha: Mocha | undefined;
```
Comment (DonJayamanne) [inline src/test/unittests.ts:31]: Even better:
```typescript
    const options: MochaSetupOptions = {
        ui: 'tdd',
        useColors: true,
        timeout,
        grep
    };

	if (MOCHA_REPORT...) { 
		options.reporter = {...}'
		options.bail = false'
	} 
	const mocha = new Mocha(options);
```

**Review by DonJayamanne — APPROVED** [review]

**Comment by DonJayamanne** [comment]
@d3r3kk where can I see the logs? Couldn't find it.

**Pull request closed without merging**